spark-rapids/spark-rapids.sh

```diff
@@ -551,44 +723,39 @@ function upgrade_kernel() {
 
   cp /var/log/dataproc-initialization-script-0.log /var/log/dataproc-initialization-script-0.log.0
 
-  systemctl reboot
+#  systemctl reboot
 }
 
 # Verify if compatible linux distros and secure boot options are used
 function check_os_and_secure_boot() {
-  if [[ "${OS_NAME}" == "debian" ]]; then
-    DEBIAN_VERSION=$(lsb_release -r | awk '{print $2}') # 10 or 11
-    if [[ "${DEBIAN_VERSION}" != "10" && "${DEBIAN_VERSION}" != "11" && "${DEBIAN_VERSION}" != "12" ]]; then
-      echo "Error: The Debian version (${DEBIAN_VERSION}) is not supported. Please use a compatible Debian version."
+  if is_debian ; then
+    if ! is_debian10 && ! is_debian11 && ! is_debian12 ; then
+      echo "Error: The Debian version ($(os_version)) is not supported. Please use a compatible Debian version."
       exit 1
     fi
-  elif [[ "${OS_NAME}" == "ubuntu" ]]; then
-    UBUNTU_VERSION=$(lsb_release -r | awk '{print $2}') # 20.04
-    UBUNTU_VERSION=${UBUNTU_VERSION%.*}
-    if [[ "${UBUNTU_VERSION}" != "18" && "${UBUNTU_VERSION}" != "20" && "${UBUNTU_VERSION}" != "22" ]]; then
-      echo "Error: The Ubuntu version (${UBUNTU_VERSION}) is not supported. Please use a compatible Ubuntu version."
+  elif is_ubuntu ; then
+    if ! is_ubuntu18 && ! is_ubuntu20 && ! is_ubuntu22 ; then
+      echo "Error: The Ubuntu version ($(os_version)) is not supported. Please use a compatible Ubuntu version."
       exit 1
     fi
-  elif [[ "${OS_NAME}" == "rocky" ]]; then
-    ROCKY_VERSION=$(lsb_release -r | awk '{print $2}') # 8 or 9
-    ROCKY_VERSION=${ROCKY_VERSION%.*}
-    if [[ "${ROCKY_VERSION}" != "8" && "${ROCKY_VERSION}" != "9" ]]; then
-      echo "Error: The Rocky Linux version (${ROCKY_VERSION}) is not supported. Please use a compatible Rocky Linux version."
+  elif is_rocky ; then
+    if ! is_rocky8 && ! is_rocky9 ; then
+      echo "Error: The Rocky Linux version ($(os_version)) is not supported. Please use a compatible Rocky Linux version."
       exit 1
     fi
   fi
 
-  if [[ "${SECURE_BOOT}" == "enabled" ]]; then 
-    echo "Error: Secure Boot is enabled. Please disable Secure Boot while creating the cluster."
+  if [[ "${SECURE_BOOT}" == "enabled" && $(echo "${DATAPROC_IMAGE_VERSION} <= 2.1" | bc -l) == 1 ]]; then 
```

Comment:
@nvliyuan - I changed the error message here.  You must be running and older version of the script.